bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -258,13 +267,73 @@ public void filter(@Nullable ContainerRequestContext requestContext) throws IOEx
                             }
                         }
                     }
-                } else if (implicitUserRole) {
+                } else if (isImplicitUserRole(requestContext)) {
                     requestContext.setSecurityContext(new AnonymousUserSecurityContext());
                 }
             } catch (AuthenticationException e) {
-                logger.warn("Unauthorized API request from {}: {}", servletRequest.getRemoteAddr(), e.getMessage());
+                logger.warn("Unauthorized API request from {}: {}", getClientIp(requestContext), e.getMessage());
                 requestContext.abortWith(JSONResponse.createErrorResponse(Status.UNAUTHORIZED, "Invalid credentials"));
             }
         }
     }
+
+    private boolean isImplicitUserRole(ContainerRequestContext requestContext) {
+        if (implicitUserRole) {
+            return true;
+        }
+        try {
+            byte[] clientAddress = InetAddress.getByName(getClientIp(requestContext)).getAddress();
+            return trustedNetworks.stream().anyMatch(networkCIDR -> networkCIDR.isInRange(clientAddress));
+        } catch (IOException e) {
+            logger.debug("Error validating trusted networks: {}", e.getMessage());
+            return false;
+        }
+    }
+
+    private List<CIDR> parseTrustedNetworks(String value) {
+        var cidrList = new ArrayList<CIDR>();
+        for (var cidrString : value.split(",")) {
+            try {
+                cidrList.add(new CIDR(cidrString.trim()));
+            } catch (UnknownHostException e) {
+                logger.warn("Error parsing trusted networks configuration, disabled.");
+                return List.of();
```

Comment:
This discards the whole list, not only this entry. Wouldn't it be better to just log the warning and continue parsing?

Reply:
Yes, I was not able to decide what to do, I will change it.